src/ameba/rule/performance/any_instead_of_empty.cr

```diff
@@ -0,0 +1,38 @@
+module Ameba::Rule::Performance
+  # This rule is used to identify usage of arg-less `any?` calls.
```

Comment:
This is not obvious why. Could you please extend the rule doc and add more explanation, so the end-user will be able to spot it while explaining the reported issue.

Reply:
Good point, I think many more rules could have their details expanded btw.

Reply:
Agree, but I think at least this one is not straightforward and may require the details to avoid questions.

Reply:
Fair enough, I'll expand the rule description.

Reply:
I've added to the documentation a section explaining the reasons behind this rule.